### PR TITLE
#9469: COG layer zoom to extent

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR adds possibility to fetch extent of COG layers

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**
- #9469 

**What is the new behavior?**
- Ability to fetch metadata of COG layer to support zoom on extent on layer
- Catalog record data model updated

### Screenshot
Screenshot 2023-09-26 at 8 08 19 PM

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/translations/data.es-ES.json [hunk 1]
```diff
@@ -1499,6 +1499,10 @@
             "title": "Catálogo",
             "tooltip": "agregar capas al mapa",
             "autoload": "Buscar en la selección de servicios",
+            "fetchMetadata": {
+                "label": "Descargar metadatos de archivos en la búsqueda",
+                "tooltip": "Esta opción recuperará metadatos para admitir el zoom a la capa al agregarla. Puede ralentizar la operación de búsqueda si las imágenes son demasiado grandes o demasiadas."
```
Comment (offtherailz) [inline web/client/translations/data.es-ES.json:1504]:
```suggestion
                "tooltip": "Esta opción recuperará metadatos para admitir el zoom a la capa. Puede ralentizar la operación de búsqueda si las imágenes son demasiado grandes o demasiadas."
```

File: web/client/translations/data.fr-FR.json [hunk 2]
```diff
@@ -1499,6 +1499,10 @@
             "title": "Catalogue",
             "tooltip": "Ajouter des couches à la carte",
             "autoload": "Recherche sur la sélection du service",
+            "fetchMetadata": {
+                "label": "Télécharger les métadonnées du fichier lors de la recherche",
+                "tooltip": "Cette option récupérera les métadonnées pour prendre en charge le zoom sur la couche lors de l'ajout. Cela peut ralentir l'opération de recherche si les images sont trop grandes ou trop nombreuses."
```
Comment (offtherailz) [inline web/client/translations/data.fr-FR.json:1504]:
```suggestion
                "tooltip": "Cette option récupérera les métadonnées pour prendre en charge le zoom sur la couche. Cela peut ralentir l'opération de recherche si les images sont trop grandes ou trop nombreuses."
```

File: web/client/translations/data.de-DE.json [hunk 3]
```diff
@@ -1537,6 +1537,10 @@
             "title": "Katalog",
             "tooltip": "Fügen Sie der Karte Ebenen hinzu",
             "autoload": "Suche in Dienstauswahl",
+            "fetchMetadata": {
+                "label": "Laden Sie Dateimetadaten bei der Suche herunter",
+                "tooltip": "Diese Option ruft Metadaten ab, um das Zoomen auf die Ebene beim Hinzufügen zu unterstützen. Es kann den Suchvorgang verlangsamen, wenn die Bilder zu groß oder zu viele sind."
```
Comment (offtherailz) [inline web/client/translations/data.de-DE.json:1542]:
```suggestion
                "tooltip": "Diese Option ruft Metadaten ab, um das Zoomen auf Ebene zu unterstützen. Es kann den Suchvorgang verlangsamen, wenn die Bilder zu groß oder zu viele sind."
```

**Review by offtherailz — CHANGES_REQUESTED** [review]
- [ ] Using this map http://localhost:8081/#/viewer/46215 (with catalog updated accordingly, given that they take a very long time, downloading something like 500MB) instead of zooming I have this errors ( and the zoom do not happen): 

```
Error in epic "zoomToExtentEpic". Original error: TypeError: Cannot read properties of null (reading 'getSimplifiedGeometry')
    at View.fit (View.js:1343:53)
    at eval (Map.jsx:316:29)
    at SwitchMapSubscriber.eval [as project] (map.js:206:7)
    at SwitchMapSubscriber._next (switchMap.js:88:27)
    at Subscriber.next (Subscriber.js:89:18)
    at FilterSubscriber._next (filter.js:88:30)
    at Subscriber.next (Subscriber.js:89:18)
    at MapSubscriber._next (map.js:83:26)
    at Subscriber.next (Subscriber.js:89:18)
    at FilterSubscriber._next (filter.js:88:30)
    at Subscriber.next (Subscriber.js:89:18)
    at WithLatestFromSubscriber._next (withLatestFrom.js:113:34)
    at Subscriber.next (Subscriber.js:89:18)
    at Subject.next (Subject.js:55:25)
[..]
EpicsUtils.js:31 Uncaught TypeError: Cannot read properties of null (reading 'getSimplifiedGeometry')
    at View.fit (View.js:1343:53)
    at eval (Map.jsx:316:29)
    at SwitchMapSubscriber.eval [as project] (map.js:206:7)
    at SwitchMapSubscriber._next (switchMap.js:88:27)
    at Subscriber.next (Subscriber.js:89:18)
    at FilterSubscriber._next (filter.js:88:30)
    at Subscriber.next (Subscriber.js:89:18)
    at MapSubscriber._next (map.js:83:26)
    at Subscriber.next (Subscriber.js:89:18)
    at FilterSubscriber._next (filter.js:88:30)
    at Subscriber.next (Subscriber.js:89:18)
    at WithLatestFromSubscriber._next (withLatestFrom.js:113:34)
    at Subscriber.next (Subscriber.js:89:18)
    at Subject.next (Subject.js:55:25)
    at eval (createEpicMiddleware.js:80:16)
    at eval (createEpicMiddleware.js:79:22)
```
Please handle this use case.

- [ ] The request really takes eons... we should open an issue to allow to cancel the catalog request while loading.
- [ ] Can we avoid to do these requests on save, but only on search, so we avoid the double time?

**Comment by dsuren1** [comment]
@offtherailz 
> * [ ]  Using this map http://localhost:8081/#/viewer/46215 (with catalog updated accordingly, given that they take a very long time, downloading something like 500MB) instead of zooming I have this errors ( and the zoom do not happen):

It caused because geokeys are empty for the layer which results in empty crs data. **Fixed**

> * [ ]  Can we avoid to do these requests on save, but only on search, so we avoid the double time?

I have added cache to avoid requests. Kindly let me know if I need to increase the expiry time, as I'm using the default expiry time now.

**Review by offtherailz — APPROVED** [review]

**Comment by offtherailz** [comment]
I think your chages are a good compromise. Thank you very much.

**Comment by ElenaGallo** [comment]
Test passed, @dsuren1 please backport. Thanks

**Pull request merged**